feat(extrafields): add extrafields hook points for enhanced customization

atm-quentin opened the pull request:
# New add extrafields hook points for enhanced customization
    - Introduced multiple new hook points (`printFieldListTitle`, `printFieldListValue`, `formObjectOptions`, `loadExtrafieldsAttributes`, etc.) to allow enhanced customization and extension of extrafields functionalities.
    - Updated templates and extrafields class to execute hooks for additional flexibility in rendering and data manipulation.
    - Ensured backward compatibility with existing extrafields implementations.

atm-quentin committed:
feat(extrafields): add hook points for enhanced customization

    - Introduced multiple new hook points (`printFieldListTitle`, `printFieldListValue`, `formObjectOptions`, `loadExtrafieldsAttributes`, etc.) to allow enhanced customization and extension of extrafields functionalities.
    - Updated templates and extrafields class to execute hooks for additional flexibility in rendering and data manipulation.
    - Ensured backward compatibility with existing extrafields implementations.

atm-quentin committed:
FIX wrong object

eldy commented on htdocs/core/class/extrafields.class.php:
```diff
@@ -1036,14 +1037,21 @@ public function fetch_name_optionals_label($elementtype, $forceload = false, $at
 		$sql = "SELECT rowid, name, label, type, size, elementtype, fieldunique, fieldrequired, param, pos, alwayseditable, perms, langs, list, printable, totalizable, fielddefault, fieldcomputed, entity, enabled, help, aiprompt,";
 		$sql .= " css, cssview, csslist";
+		$parameters = ['elementtype' => $elementtype, 'forceload' => $forceload, 'attrname' => $attrname];
+		$hookmanager->executeHooks('printExtrafieldsSelect', $parameters, $this, $action);
```
The goal of fetch_name_optionals_label is to load the definition of the system table extrafields.
There is no customization of this system table, because this table contains data to customize the business object but is not customizable itself.

Can you describe your use case why you need extrafields on extrafields table ?

atm-quentin replied:
for a customer we want to add specific fields on extrafields
we made a specific join table, but we need to load thoses fields in extrafield list / view

eldy replied:
Sorry i don't understand what you mean with adding extrafields on extrafields.
Can you describe your use case why you need extrafields on extrafields table ?

atm-quentin committed:
feat(extrafields): add hooks for enhanced CRUD customization

- Added new hook points (`addExtraField`, `updateExtrafields`, `afterUpdateExtraField`, etc.) for additional flexibility in extrafields' CRUD operations.
- Refactored query generation in extrafields class to support hook-based customizations and ensure proper scoping.
- Enhanced support for managing extrafields attributes dynamically via hooks.
